src/video/video_sdl.c

```diff
@@ -470,6 +487,116 @@ static void Video_DrawScreen(void)
 	SDL_UnlockSurface(s_gfx_surface);
 }
 
+#if defined(_WIN32) && defined(WITH_SDL)
+/**
+* Check if Window can be resized
+* @param new_magnification The new magnification to check.
+*/
+static bool Video_CanResize(int32 new_magnification)
+{
+	if ((SCREEN_WIDTH  * new_magnification >= s_desktop_width || SCREEN_HEIGHT * new_magnification >= s_desktop_height) ||
+		new_magnification <= 0) return false;
+
+	return true;
+}
+
+/**
+* Resize SDL_SetVideoMode on Win32.
+*/
+static void Video_Resize(void)
+{
+	SDL_Color colors[256];
+	if (g_scale_filter != FILTER_HQX) {
+		for (int i = 0; i < 256; i++) {
+			colors[i].r = s_gfx_surface->format->palette->colors[i].r;
+			colors[i].g = s_gfx_surface->format->palette->colors[i].g;
+			colors[i].b = s_gfx_surface->format->palette->colors[i].b;
+		}
+	}
+
+	s_gfx_surface = SDL_SetVideoMode(SCREEN_WIDTH * g_screen_magnification, SCREEN_HEIGHT * g_screen_magnification, s_bpp, s_surfaceFlags);
+	if (s_gfx_surface == NULL) {
+		Warning("Failed to toggle full screen\n");
+	} else if (g_scale_filter != FILTER_HQX) {
+		//SDL_SetPalette(s_gfx_surface, SDL_LOGPAL | SDL_PHYSPAL, colors, 0, 256);
+		if (!SDL_SetPalette(s_gfx_surface, SDL_LOGPAL | SDL_PHYSPAL, colors, 0, 256)) Warning("SDL_SetPalette() failed while resizing\n");
+	}
+}
+
+/**
+* Toggle the Fullscreen mode on Win32.
+*/
+static void Video_ToggleFullscreen(void)
+{
+	if (!s_fullscreen) {
+		s_prev_magnification = g_screen_magnification;
+		g_screen_magnification = 2;
+		
```

Comment:
Here we fix magnification to 2 because of HQX filter breaks surface pixels filling, and using a real fullscreen mode it can get better aspect ratio dimensions from 2x instead of from 3x.